design/specs/table-import.md

```diff
@@ -14,153 +14,175 @@ Adding a new table from a file import allows users to input data into Mathesar w
+This consists of the file the user uploaded. They can access it from here when they click on it. If the reupload button is selected it has a similar behavior like go back button.
 
+#### Table name
 
+The input has a default name that can be changed by user there itself.
+
+#### Import Summary (Total Records, Columns)
+
+This has a small summary like number of rows, records etc. The number of missing data and duplicated columns is present on the confirmation view. Although user can ignore those warnings and table can be confirmed as is.
+
+#### Options
+
+- Use headers - This will use the first row as headers if selected or else there will be default column names. This will remove duplicate column scenario.
+- Set missing data as null - This will set missing data as null and remove missing data scenario.
+
+#### Verifying and Updating Data Types
+
+*Note - The UX of changing data types depends on table inspector designs.*
+
+This is similar to the data options menu for the table view. But there are small differences to remove Cognitive load on user -
+
+1. Removal of options like filtering, sorting etc since it might not be that useful to user at this point.
+2. Removal of options which are related to adding new data like set default value. Since we are not allowing user to add data at this stage keeping these options can be avoided. However all the options related to the data type and data format will be intact since this might be a deciding factor. 
+
+##### Allowed Data Type Changes
+
+The data type changes behavior will be similar to that of table data type changes. User will be able to switch to compatible data types so they do not lose the data.
+
+##### Revert Data Type Change
+
+User can change the data types to the other compatible ones at any moment in the confirmation stage. 
+
+#### Exclude Columns from Import
+
+Each column has a checkmark on the side. If selected the table will have the column or else it will be excluded from the table. 
+
+All the columns will be selected by default including the duplicated columns. The duplicate columns will be highlighted with a warning though.
+
+#### Abandoning the Confirmation Step
+
+![](https://share.balsamiq.com/c/ajPCXuhshEjRMvJXMLb5Xu.png)
+
+Handling unconfirmed table in different scenarios might be difficult so the user will no be allowed to navigate away without interacting with a modal. 
+
+In the case of force close, the table listed on the schema page will be highlighted with a warning. Any attempt to use an unconfirmed table will lead user to a modal to either confirm the table as is or go to the confirmation view.
+
+## UI Components
+
+### File Upload Method Selector
+
+![](https://share.balsamiq.com/c/jyszUUBfSvoCqd6qWQcz3G.png)
+
+To select the method to upload the file there will be vertical tabs to select the method. There will be a rectangular space attached to the tabs. This space will be dedicated to the upload method selected. 
```

Comment:
Please see note about vertical tabs above – they don't make sense for a mutually exclusive choice.